### ISSUE TITLE
Kafka consumer offset

### ISSUE THREAD

**Issue by salekd** [AstroPlant/astroplant-backend#23]
Make sure we don't always start from the latest message.

**Comment by salekd** [comment]
`auto_offset_reset='earliest'` and `enable_auto_commit=False` should ensure consuming from the first message stored in Kafka.

**Comment by salekd** [comment]
I am not sure we want to always start consuming from the first/earliest message. We are talking about restarting a consumer and I believe we should always start where we stopped before. The defaults settings below should already handle this:
`enable_auto_commit=True`
`auto_commit_interval_ms=5000`

@Beskhue what was the behaviour you observed? Did you say the consumer always started from the current time, i.e. all past messages were lost?

**Comment by tomcur** [comment]
> @Beskhue what was the behaviour you observed? Did you say the consumer always started from the current time, i.e. all past messages were lost?

I observed the same two behaviours as you describe: (1) without `auto_offset_reset` set (default is `latest`), consumers only see messages sent after their (re)connection. And (2) with `auto_offset_reset` at `earliest`, consumers see all messages every time upon reconnection, even if they've already seen it before.

I've not tested with consumers in consumer groups yet, but thought/hoped with consumer groups set up, situation (2) would mean consumers in the group would see exactly those messages as yet unconsumed for their group, be they old or new, so "we should always start where we stopped before." That's the desired bahaviour.
